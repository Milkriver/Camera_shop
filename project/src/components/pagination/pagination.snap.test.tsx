import {render} from '@testing-library/react';
import { createMemoryHistory } from 'history';
import HistoryRouter from '../history-router/history-router';
import Pagination from './pagination';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

const history = createMemoryHistory();
const mockPages = [1,2,3];

describe('Component: Pagination', () => {
  it('should render correctly', () => {
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);

    const store = mockStore({
        DATA: {
            activePaginationPage: 1,
        }
    });
    const {container} = render(
        <Provider store={store}>
            <HistoryRouter history={history}>
                <Pagination pages={mockPages}/>
            </HistoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
