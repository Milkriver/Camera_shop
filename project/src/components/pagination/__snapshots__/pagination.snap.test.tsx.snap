// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Pagination should render correctly 1`] = `
<div>
  <ul
    class="pagination__list"
  >
    <li
      class="pagination__item"
      style="cursor: pointer;"
    >
      <div
        class="pagination__link pagination__link--active"
      >
        1
      </div>
    </li>
    <li
      class="pagination__item"
      style="cursor: pointer;"
    >
      <div
        class="pagination__link "
      >
        2
      </div>
    </li>
    <li
      class="pagination__item"
      style="cursor: pointer;"
    >
      <div
        class="pagination__link "
      >
        3
      </div>
    </li>
    <li
      class="pagination__item"
      style="cursor: pointer;"
    >
      <div
        class="pagination__link pagination__link--text"
      >
        Далее
      </div>
    </li>
  </ul>
</div>
`;
